crates/core/src/factories/core_fetch.rs

```diff
@@ -173,12 +172,15 @@ impl Fetch for CoreFetch {
             // These need to be added up front, before sending them to the outgoing
             // request queue, otherwise the queue processes them faster than they're
             // being added to state and the request logic fails.
-            self.state.lock().unwrap().requests.extend(
-                new_op_ids
-                    .clone()
-                    .into_iter()
-                    .map(|op_id| (op_id.clone(), source.clone())),
-            );
+            {
+                let requests = &mut self.state.lock().unwrap().requests;
+                requests.extend(
+                    new_op_ids
+                        .clone()
+                        .into_iter()
+                        .map(|op_id| (op_id.clone(), source.clone())),
+                );
+            }
```

Comment:
This is functionally equivalent I believe but makes reading it clearer that we're holding a lock for this block only.